pkg/operator/controller/controller_router_deployment.go

```diff
@@ -172,8 +172,11 @@ func desiredRouterDeployment(ci *ingressv1alpha1.ClusterIngress, routerImage str
 		})
 	}
 
-	replicas := ci.Spec.Replicas
-	deployment.Spec.Replicas = &replicas
+	var desiredReplicas int32 = 2
+	if ci.Spec.Replicas != nil {
+		desiredReplicas = *ci.Spec.Replicas
+	}
+	deployment.Spec.Replicas = &desiredReplicas
```

Comment:
Subtle but important difference from prior behavior here, whereby we defaulted `.spec.replicas` (which wasn't a pointer)

Reply:
Since the defaulting logic is now in `desiredRouterDeployment`, we should add a check to `TestDesiredRouterDeployment` where `ci.Spec.Replicas` is nil.